shared/js/sidebar.jsx

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var loop = loop || {};
+
+loop.sidebar = (function(mozL10n) {
+  "use strict";
+
+  var ROOM_STATES = loop.store.ROOM_STATES;
+  var sharedActions = loop.shared.actions;
+  var sharedUtils = loop.shared.utils;
+  var sharedMixins = loop.shared.mixins;
+
+  var SidebarControllerView = React.createClass({
+    mixins: [sharedMixins.UrlHashChangeMixin,
+             sharedMixins.DocumentLocationMixin,
+             loop.store.StoreMixin("activeRoomStore"),
+             Backbone.Events],
+
+    propTypes: {
+      activeRoomStore: React.PropTypes.instanceOf(loop.store.ActiveRoomStore).isRequired,
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired
+    },
+
+    getInitialState: function() {
+      return this.getStoreState();
+    },
+
+    componentWillMount: function() {
+      this.props.activeRoomStore.on("change", function() {
+        this.setState(this.props.activeRoomStore.getStoreState());
+      }, this);
+    },
+
+    componentWillUnmount: function() {
+      this.props.activeRoomStore.off("change", null, this);
+    },
+
+    componentWillUpdate: function(nextProps, nextState) {
+      console.info(this.state, nextState);
+    },
+
+    render: function() {
+      if (this.state.roomState === ROOM_STATES.ROOM_GATHER) {
+        // The state hasn't been initialised yet, so don't display
+        // anything to avoid flicker.
+        return (
+          <div>
+            <p>{"If you see this, please file a bug"}</p>
+          </div>
+        );
+      }
+
+      return (
+         <DesktopSidebarView
+           activeRoomStore={this.props.activeRoomStore}
+           dispatcher={this.props.dispatcher} />
+      );
+    }
+  });
+
+  var DesktopSidebarView = React.createClass({
+    propTypes: {
+      activeRoomStore: React.PropTypes.instanceOf(loop.store.ActiveRoomStore).isRequired,
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired
+    },
+
+    getInitialState: function() {
+      return this.props.activeRoomStore.getStoreState();
```

Comment:
This view isn't listening to changes. So its likely it isn't going to update correctly. Preferably, it should just use the store mixin.